### ISSUE TITLE
cdc do_update_streams_description error: Can't find a column family with UUID fdf455c4-cfec-3e00-9719-d7a45436c89d

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#9287]
Seen in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/921/testReport/junit/upgrade_test/UpgradeTestFrom40ToLast/test_cluster_upgrade/
```
Unexpected errors found: [('node1', ["ERROR 2021-09-05 05:24:01,437 [shard 0] storage_proxy - Exception when communicating with 127.0.83.3, to read from system_distributed.cdc_generation_timestamps: std::runtime_error (Can't find a column family with UUID fdf455c4-cfec-3e00-9719-d7a45436c89d)"])]
```

It looks like cdc updates streams description very early in the start up process, before all other nodes complete their setup of the `system_distributed.cdc_generation_timestamps` table.

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/921/artifact/logs-all.release.2/1630819714727_upgrade_test.UpgradeTestFrom40ToLast.test_cluster_upgrade/node1.log
```
Scylla version 4.0.9-0.20200917.d86a31097a with build-id 8f7e552bb4ebb84b54b21cc04c333e6b6771bea7 starting ...
...
INFO  2021-09-05 05:24:01,278 [shard 0] migration_manager - Create new ColumnFamily: org.apache.cassandra.config.CFMetaData@0x600003b2ce00[cfId=fdf455c4-cfec-3e00-9719-d7a45436c89d,ksName=system_distributed,cfName=cdc_generation_timestamps,cfType=Standard,comparator=org.apache.cassandra.db.marshal.CompositeType(org.apache.cassandra.db.marshal.ReversedType(org.apache.cassandra.db.marshal.TimestampType),org.apache.cassandra.db.marshal.UTF8Type),comment=,readRepairChance=0,dcLocalReadRepairChance=0,gcGraceSeconds=864000,keyValidator=org.apache.cassandra.db.marshal.UTF8Type,minCompactionThreshold=4,maxCompactionThreshold=32,columnMetadata=[ColumnDefinition{name=key, type=org.apache.cassandra.db.marshal.UTF8Type, kind=PARTITION_KEY, componentIndex=0, droppedAt=-9223372036854775808}, ColumnDefinition{name=time, type=org.apache.cassandra.db.marshal.ReversedType(org.apache.cassandra.db.marshal.TimestampType), kind=CLUSTERING_COLUMN, componentIndex=0, droppedAt=-9223372036854775808}, ColumnDefinition{name=expired, type=org.apache.cassandra.db.marshal.TimestampType, kind=REGULAR, componentIndex=null, droppedAt=-9223372036854775808}],compactionStrategyClass=class org.apache.cassandra.db.compaction.SizeTieredCompactionStrategy,compactionStrategyOptions={enabled=true},compressionParameters={sstable_compression=org.apache.cassandra.io.compress.LZ4Compressor},bloomFilterFpChance=0.01,memtableFlushPeriod=0,caching={"keys":"ALL","rows_per_partition":"ALL"},cdc={},defaultTimeToLive=0,minIndexInterval=128,maxIndexInterval=2048,speculativeRetry=99.0PERCENTILE,triggers=[],isDense=false,version=580abf0b-aadf-326c-ae0e-6470ef4bbb37,droppedColumns={},collections={},indices={}]
INFO  2021-09-05 05:24:01,297 [shard 1] compaction - [Compact system_schema.scylla_tables 7a6b0010-0e09-11ec-a7a7-000000000001] Compacting [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/scylla_tables-5d912ff1f7593665b2c88042ab5103dd/md-149-big-Data.db:level=0, /jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/scylla_tables-5d912ff1f7593665b2c88042ab5103dd/md-147-big-Data.db:level=0, ]
INFO  2021-09-05 05:24:01,321 [shard 1] compaction - [Compact system_schema.columns 7a65f700-0e09-11ec-a7a7-000000000001] Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/columns-24101c25a2ae3af787c1b40ee1aca33f/md-151-big-Data.db:level=0, ]. 78kB to 72kB (~92% of original) in 33ms = 2MB/s. ~256 total partitions merged to 5.
INFO  2021-09-05 05:24:01,370 [shard 1] compaction - [Compact system_schema.dropped_columns 7a7623a0-0e09-11ec-a7a7-000000000001] Compacting [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/dropped_columns-5e7583b5f3f43af19a39b7e1d6f5f11f/md-71-big-Data.db:level=0, /jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/dropped_columns-5e7583b5f3f43af19a39b7e1d6f5f11f/md-69-big-Data.db:level=0, /jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/dropped_columns-5e7583b5f3f43af19a39b7e1d6f5f11f/mc-67-big-Data.db:level=0, ]
INFO  2021-09-05 05:24:01,381 [shard 0] schema_tables - Creating system_distributed.cdc_generation_timestamps id=fdf455c4-cfec-3e00-9719-d7a45436c89d version=5777e043-a81a-3324-8803-b497c056849f
INFO  2021-09-05 05:24:01,382 [shard 1] compaction - [Compact system_schema.scylla_tables 7a6b0010-0e09-11ec-a7a7-000000000001] Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/scylla_tables-5d912ff1f7593665b2c88042ab5103dd/md-153-big-Data.db:level=0, ]. 68kB to 63kB (~92% of original) in 73ms = 867kB/s. ~256 total partitions merged to 5.
INFO  2021-09-05 05:24:01,385 [shard 1] compaction - [Compact system_schema.scylla_tables 7a786d90-0e09-11ec-a7a7-000000000001] Compacting [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/scylla_tables-5d912ff1f7593665b2c88042ab5103dd/md-151-big-Data.db:level=0, /jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/scylla_tables-5d912ff1f7593665b2c88042ab5103dd/md-153-big-Data.db:level=0, ]
INFO  2021-09-05 05:24:01,389 [shard 0] schema_tables - Schema version changed to 1865da7e-a742-3131-a453-0fdf0989ed23
INFO  2021-09-05 05:24:01,390 [shard 0] storage_service - Using saved tokens {minimum token}
INFO  2021-09-05 05:24:01,398 [shard 1] compaction - [Compact system_schema.dropped_columns 7a7623a0-0e09-11ec-a7a7-000000000001] Compacted 3 sstables to [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/dropped_columns-5e7583b5f3f43af19a39b7e1d6f5f11f/md-73-big-Data.db:level=0, ]. 83kB to 39kB (~47% of original) in 15ms = 2MB/s. ~384 total partitions merged to 2.
INFO  2021-09-05 05:24:01,405 [shard 0] compaction - [Compact system.cdc_local 7a7b7ad0-0e09-11ec-be5a-000000000000] Compacting [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system/cdc_local-0bcaffd40c833eadad13dc1d5015b77c/md-4-big-Data.db:level=0, /jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system/cdc_local-0bcaffd40c833eadad13dc1d5015b77c/md-2-big-Data.db:level=0, ]
INFO  2021-09-05 05:24:01,410 [shard 1] compaction - [Compact system_schema.keyspaces 7a7c3e20-0e09-11ec-a7a7-000000000001] Compacting [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/keyspaces-abac5682dea631c5b535b3d6cffd0fb6/md-165-big-Data.db:level=0, /jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/keyspaces-abac5682dea631c5b535b3d6cffd0fb6/md-163-big-Data.db:level=0, /jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/keyspaces-abac5682dea631c5b535b3d6cffd0fb6/md-161-big-Data.db:level=0, ]
INFO  2021-09-05 05:24:01,424 [shard 1] compaction - [Compact system_schema.scylla_tables 7a786d90-0e09-11ec-a7a7-000000000001] Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/scylla_tables-5d912ff1f7593665b2c88042ab5103dd/md-155-big-Data.db:level=0, ]. 68kB to 63kB (~92% of original) in 25ms = 2MB/s. ~256 total partitions merged to 5.
INFO  2021-09-05 05:24:01,435 [shard 0] storage_service - Node 127.0.83.1 state jump to normal
INFO  2021-09-05 05:24:01,436 [shard 0] compaction - [Compact system.local 7a8035c0-0e09-11ec-be5a-000000000000] Compacting [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system/local-7ad54392bcdd35a684174e047860b377/md-96-big-Data.db:level=0, /jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system/local-7ad54392bcdd35a684174e047860b377/md-94-big-Data.db:level=0, ]
INFO  2021-09-05 05:24:01,436 [shard 1] compaction - [Compact system_schema.tables 7a8035c0-0e09-11ec-a7a7-000000000001] Compacting [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/tables-afddfb9dbc1e30688056eed6c302ba09/md-145-big-Data.db:level=0, /jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/tables-afddfb9dbc1e30688056eed6c302ba09/md-143-big-Data.db:level=0, /jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-xi8o1mtw/test/node1/data/system_schema/tables-afddfb9dbc1e30688056eed6c302ba09/md-141-big-Data.db:level=0, ]
ERROR 2021-09-05 05:24:01,437 [shard 0] storage_proxy - Exception when communicating with 127.0.83.3, to read from system_distributed.cdc_generation_timestamps: std::runtime_error (Can't find a column family with UUID fdf455c4-cfec-3e00-9719-d7a45436c89d)
INFO  2021-09-05 05:24:01,445 [shard 0] cdc - CDC description table successfully updated with generation 2021/09/05 05:23:49.
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/921/artifact/logs-all.release.2/1630819714727_upgrade_test.UpgradeTestFrom40ToLast.test_cluster_upgrade/node2.log
```
INFO  2021-09-05 05:24:01,413 [shard 0] schema_tables - Creating system_distributed.cdc_generation_timestamps id=fdf455c4-cfec-3e00-9719-d7a45436c89d version=5777e043-a81a-3324-8803-b497c056849f
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/921/artifact/logs-all.release.2/1630819714727_upgrade_test.UpgradeTestFrom40ToLast.test_cluster_upgrade/node3.log
```
INFO  2021-09-05 05:24:01,455 [shard 0] schema_tables - Creating system_distributed.cdc_generation_timestamps id=fdf455c4-cfec-3e00-9719-d7a45436c89d version=5777e043-a81a-3324-8803-b497c056849f
```

**Comment by haaawk** [comment]
Is CDC enabled as experimental on 4.0?

**Comment by kbr-** [comment]
The only issue here is an error log message, it doesn't cause any actual problems to users of CDC. The caller of `do_update_streams_description` expects that the call may sometimes fail and will retry until it succeeds. In the case we see here it didn't even fail because 2 out of 3 replicas successfully performed the read so the entire quorum read succeeded even though one replica failed.

The cause is that replicas wouldn't perform schema sync on reads, they would only do it on writes, but `system_distributed_keyspace::cdc_desc_exists` (called from `do_update_streams_description`) performs a read. The problem is fixed in newer versions of Scylla (c16e4a04239798c41fa140c14eabb250b5422983) but apparently the commit is not backported to 4.3, and even if it was it wouldn't help users who upgrade from minor versions of 4.3 which don't have the backport. (for this bug it doesn't matter which version you upgrade to, it matters which version you upgrade from)

I think the best "fix" is to document this behavior, i.e. that users may sometimes see this message when upgrading from 4.3. If they see further down in the logs that
```
[shard 0] cdc - CDC description table successfully updated with generation ...
```
then there is nothing to worry about.
